src/main/kotlin/com/ampnet/mailservice/service/impl/mail/AbstractMail.kt

```diff
@@ -20,22 +23,22 @@ abstract class AbstractMail(
     protected val linkResolver: LinkResolverService,
     private val mailSender: JavaMailSender,
     private val applicationProperties: ApplicationProperties,
-    private val translationService: TranslationService
+    private val headlessCmsService: HeadlessCmsService
 ) {
 
     companion object : KLogging()
 
-    protected abstract val templateName: String
-    protected abstract val titleKey: String
-    protected lateinit var language: String
+    protected abstract val mailType: MailType
+    protected lateinit var language: Lang
+    protected lateinit var coop: String
     protected open var attachment: Attachment? = null
     protected open var templateData: Any? = null
-    private val templateTranslations: Map<String, Mustache> by lazy {
-        translationService.getTemplateTranslations(templateName)
+
+    fun setLanguage(language: String) = apply {
+        this.language = Lang.langOrDefault(language)
```

Comment:
The app would break if the setLanguage is not called. However, language cannot be undefined, it's fetched through grpc, so it can only be an empty string ("") which would then fallbackt to Lang.EN. But good suggestion I will put the default value for language in Abstract mail to Lang.EN. value.

Reply:
Ok, the only problem is with the value `undefined` for late init.